scripts/firedrake-install

```diff
@@ -407,30 +422,33 @@ If you really want to use your own Python packages, please run again with the
 
 petsc_opts = "--download-eigen=%s/src/eigen-3.3.3.tgz " % firedrake_env
 
-petsc_opts += "--with-fortran-bindings=0 "
+full_opts = '--download-chaco --download-metis --download-parmetis \
+--download-scalapack --download-hypre --download-mumps --with-zlib --download-netcdf --download-pnetcdf --download-hdf5 --download-exodusii --with-fortran-bindings=0'
 
-if options["minimal_petsc"]:
-    if options["petsc_int_type"] == "int64":
-        petsc_opts += """--download-metis --download-parmetis --download-hdf5 --download-hypre --with-64-bit-indices"""
-    else:
-        petsc_opts += """--download-chaco --download-metis --download-parmetis --download-scalapack --download-hypre --download-mumps --download-hdf5"""
-else:
-    if options["petsc_int_type"] == "int64":
-        petsc_opts += """--download-metis --download-parmetis --download-hypre --with-zlib --download-netcdf --download-pnetcdf --download-hdf5 --download-exodusii --with-64-bit-indices"""
-    else:
-        petsc_opts += """--download-chaco --download-metis --download-parmetis --download-scalapack --download-hypre --download-mumps --with-zlib --download-netcdf --download-hdf5 --download-pnetcdf --download-exodusii"""
+split_opts = set(full_opts.split())
 
+if options["minimal_petsc"]:
+    split_opts -= {'--with-zlib', '--download-netcdf', '--download-pnetcdf', '--download-exodusii'}
+if options["petsc_int_type"] == 'int64':
+    split_opts -= {'--download-chaco', '--download-scalapack', '--download-mumps'}
+    split_opts |= {'--with-64-bit-indices'}
+if options["complex"]:
+    split_opts -= {'--download-hypre'}
+    split_opts |= {'--with-scalar-type=complex'}
 
 if "PETSC_CONFIGURE_OPTIONS" not in os.environ:
-    os.environ["PETSC_CONFIGURE_OPTIONS"] = petsc_opts
+    os.environ["PETSC_CONFIGURE_OPTIONS"] = " ".join(split_opts)
```

Comment:
Although we say this in `--help`, it's somehow not matched up with everything else (where you pass stuff as an option). Do we want to change `firedrake-install` such that you say `firedrake-install --additional-petsc-options ...` instead of setting an environment variable?

Reply:
Finally, I added a `--show-petsc-configure-options` command to `firedrake-install` a while back, but it doesn't exit early (so we can't use it to show what you need to pass to petsc's configure). Should we do that properly while we're messing around here?

Should we also be explicit about all the options that petsc's setup.py puts in? https://github.com/firedrakeproject/petsc/blob/firedrake/setup.py#L94

Reply:
Is there a clean way to extract the options from petsc's setup.py? I don't particularly just want to copy the options into Firedrake...

Reply:
you can run `config` with dry_run=True and it prints them out and we could get that.

OTOH, if we do that we should go ahead I think and manage the petsc install ourselves.